core/modules/book/book.module

```diff
@@ -659,6 +666,7 @@ function _book_update_outline(Node $node) {
         // Update the bid for this page and all children.
         book_update_bid($node->book);
         // Reset the cache of stored books.
+        _book_cache_clear($node->book['bid'], TRUE);
```

Comment:
Both invocations: `_book_cache_clear()` and `backdrop_static_reset()` will happen regardless of the value of `$new` in this `if / else` block. Therefore these two can be moved outside the `if / else`.